### PR TITLE
Bug 1904684: get rid of docker.io in tests

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

File: test/extended/testdata/cmd/test/cmd/newapp.sh
Comment (adambkaplan) [inline test/extended/testdata/cmd/test/cmd/newapp.sh:461]: doesn't this still use docker.io?

**Review by adambkaplan — COMMENTED** [review]
Generally looks good.

For wildfly replacements, `quay.io/wildfly/wildfly-centos-7` should fill the gap.

File: test/extended/testdata/cmd/test/cmd/newapp.sh
Comment (soltysh) [reply test/extended/testdata/cmd/test/cmd/newapp.sh:461]: It does :( There are a few sill left, but I can't find anything to replace that temporarily. I'll return to this after this merges.

Comment (soltysh) [reply test/extended/testdata/cmd/test/cmd/newapp.sh:461]: Also newapp.sh is on the blacklist https://github.com/openshift/origin/blob/ee82ae0d236efb69590dd4ed5ad29452eb62a4a0/test/extended/cmd/cmd.go#L28-L34 so it's not executed.

Comment (adambkaplan) [reply test/extended/testdata/cmd/test/cmd/newapp.sh:460]: language nit - we should rename this var to `skipList`. The usage of blacklist/whitelist is discouraged.

Comment (soltysh) [reply test/extended/testdata/cmd/test/cmd/newapp.sh:460]: This will disappear as soon as I'm done with all the migration.

**Comment by adambkaplan** [comment]
LGTM

@soltysh do you plan on squashing commits?

**Comment by soltysh** [comment]
> @soltysh do you plan on squashing commits?

Nope, I'm planning on re-freshing this PR a bit after the dust settles from k8s bump landing.

**Comment by jaypoulz** [comment]
@soltysh is there a way to use images that are UBI based instead of centos?
Centos is not built for s390x, any images based on it will not be compatible for multi-arch tests.

**Comment by soltysh** [comment]
> @soltysh is there a way to use images that are UBI based instead of centos?
> Centos is not built for s390x, any images based on it will not be compatible for multi-arch tests.

I'm fine with doing that as a follow-up, but I'll leave that up to you. Ping me for review, though.

**Review by adambkaplan — APPROVED** [review]
/lgtm

**Comment by yselkowitz** [comment]
A number of the files modified in this commit are meant to be synced automatically from openshift/library, meaning that these changes will be lost on the next sync.  The better approach -- presuming both are similarly up to date -- would be to modify the imagestreams at their source to use registry.centos.org instead of docker.io, which would then benefit both OKD users as well as future syncs here.

As for using UBI, that's already being used in imagestreams where available.  However, the databases are not included in UBI and therefore that's not an option for these.

**Comment by dmage** [comment]
It's a 4.7.0 fix, and I want to have it in 4.6.z, but I don't want to do it manually :) I don't think 4.6.z is synced with anything.

**Comment by yselkowitz** [comment]
This was just merged to master though.

**Comment by dmage** [comment]
But it was merged a year ago :)

**Comment by soltysh** [comment]
Yeah, it's not that simple, we'll have to pick it up manually, I'm hoping to get to it tomorrow.

**Pull request merged**
